includes/Admin/Admin_AJAX.php

```diff
@@ -221,8 +228,9 @@ public function get_checks_to_run() {
 
 		wp_send_json_success(
 			array(
-				'plugin' => $plugin_basename,
-				'checks' => array_keys( $checks_to_run ),
+				'plugin'     => $plugin_basename,
+				'checks'     => array_keys( $checks_to_run ),
+				'categories' => $categories,
```

Comment:
This change should be unnecessary, the categories shouldn't be returned from here. The purpose of the `get_checks_to_run()` AJAX callback is to return the full list of checks to run, which should already be filtered by the categories per your logic added to `Abstract_Check_Runner::get_checks_to_run()`. So returning the categories isn't needed.

Instead, in the `try catch` clause right above here, you need to set the check categories in the runner via the method for that (which currently has the misleading name `filter_checks_by_specific_categories`, it should really be called `set_categories`, as mentioned in my other comment).